packages/extension/src/utils/format.ts

```diff
@@ -93,3 +93,28 @@ export const formatFlagsToNumber = (tx: Transaction) => {
 export const formatTransferFee = (fee: number) => {
   return fee / 1000;
 };
+
+// If the given string is longer than the given length, and does not contain a space within the length,
+// then it will be truncated and an ellipsis will be added to the end.
+// We do this, because if there is a space within the range, the string will be correctly displayed in the UI.
+export const formatStringToBeDisplayed = (str: string, length: number) => {
```

Comment:
This is in the util folder, the function doesn't have a generic explicit name, I would rename it, not sure to understand what it does without the help of the comments and reading all of it.

Reply:
Here is what the function does

When we display some strings to the UI, we want to make sure that they are not too long and will not break the UI.
To do so, we truncate the string only if:
 - There is one word with a length bigger than the given length
 otherwise, we return the string as is.  
 
We do not need to truncate the string if there is no word with a length bigger than the given length because in this
 case, the string will be correctly displayed in multiple lines, and will not break the UI.

Reply:
![Uploading Screenshot 2023-08-06 at 22.10.05.png…]()

Reply:
Here is a case where the string (NFT description) is longer than the length of the UI, but it is not truncated because it does not contain long words, so it does not break the UI. If it contained a long word, such as a NFT Token ID, it would break the UI.   

<img width="358" alt="Screenshot 2023-08-06 at 22 10 05" src="https://github.com/GemWallet/gemwallet-extension/assets/9871294/f9a90f46-1486-46ea-b844-21ade95dc579">

Reply:
I have found a way to do without this function, with a new component called <TruncatedText> 👍